### PR TITLE
Lint evaluation, lazy and meta

### PR DESCRIPTION
**Pull request by garawalid**
Lint `meta`, `lazy` and `evaluation`. Part of #219.

### CODE REVIEW

**Comment by garawalid** [comment]
I'll rebase this PR once #262 merged!

File: src/skmultiflow/data/time_manager.py [hunk 1]
```diff
@@ -1,156 +1,156 @@
...
-        # check if sample_weight are being used
-        if self._USE_SAMPLE_WEIGHT:
-            sample_weight = np.array(samples["weight"].values.tolist())
-        else:
-            sample_weight = None
-        # remove samples from queue
-        self._cleanup_samples(batch_size)
-        # return X, y_true, y_pred, and weight for the dequeued samples
-        return X, y_true, y_pred, sample_weight
+import numpy as np
```
Comment (garawalid) [inline src/skmultiflow/data/time_manager.py:1]: I'll fix `time_manager.py` later with the rebase!

**Comment by garawalid** [comment]
@smastelini  if you have some free time, would you mind please review this PR? 
Just focus on `meta`, `lazy` and `evaluation`. 
Thanks!

**Comment by smastelini** [comment]
Hi Walid, I just merged #262. I will proceed to this one. Could please you sync this branch with `upstream/master` so that I can focus only on the actual scope of this PR?

**Comment by garawalid** [comment]
Thanks @smastelini ! This PR is ready for the review.

File: src/skmultiflow/evaluation/base_evaluator.py [hunk 2]
```diff
@@ -607,80 +678,137 @@ def evaluation_summary(self):
...
+                        self.model_names[i],
+                        self._data_buffer.get_data(
+                            metric_id=constants.RUNNING_TIME,
+                            data_id='testing_time')[i]))
+                print(
+                    '{} - Total time    (s)  : {:.2f}'.format(
+                        self.model_names[i],
+                        self._data_buffer.get_data(
+                            metric_id=constants.RUNNING_TIME,
+                            data_id='total_running_time')[i]))
```
Comment (smastelini) [inline src/skmultiflow/evaluation/base_evaluator.py:811]: Although technically consistent with PEP8, these print commands on which the content only start in the next line look odd to me. I suggest formatting them in a different way. Perhaps keeping the characters till `.format(` in the first line would be a solution.

File: src/skmultiflow/meta/classifier_chains.py [hunk 3]
```diff
@@ -27,6 +27,7 @@ class ClassifierChain(BaseSKMObject, ClassifierMixin, MetaEstimatorMixin, MultiO
     Examples
     --------
     >>> from skmultiflow.data import make_logical
+    >>> from sklearn.linear_model import SGDClassifier
```
Comment (smastelini) [inline src/skmultiflow/meta/classifier_chains.py:30]: Good catch!

File: src/skmultiflow/meta/classifier_chains.py [hunk 4]
```diff
@@ -263,10 +268,11 @@ def P(y, x, cc, payoff=np.prod):
...
     xy[0:D] = x.copy()
     for j in range(L):
-        P_j = cc.ensemble[j].predict_proba(xy[0:D+j].reshape(1, -1))[0]  # e.g., [0.9, 0.1] wrt 0, 1
-        xy[D+j] = y[j]                                           # e.g., 1
+        P_j = cc.ensemble[j].predict_proba(
+            xy[0:D + j].reshape(1, -1))[0]  # e.g., [0.9, 0.1] wrt 0, 1
+        xy[D + j] = y[j]                                           # e.g., 1
         p[j] = P_j[y[j]]                                         # e.g., 0.1
-                                                                 # or, y[j] = 0 is predicted with probability p[j] = 0.9
+        # or, y[j] = 0 is predicted with probability p[j] = 0.9
```
Comment (smastelini) [inline src/skmultiflow/meta/classifier_chains.py:275]: Just a minor detail, I am not sure of its importance. Is it possible to align the comments (as they were before) without breaking the length limit?

File: src/skmultiflow/meta/online_boosting.py [hunk 5]
```diff
@@ -193,7 +201,8 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
             if set(self.classes) == set(classes):
                 pass
             else:
-                raise ValueError("The classes passed to the partial_fit function differ from those passed earlier.")
+                raise ValueError("The classes passed to the partial_fit function differ "
+                                 "from those passed earlier.")
```
Comment (smastelini) [inline src/skmultiflow/meta/online_boosting.py:205]: What about `if set(self.classes) != set(classes):`?

File: src/skmultiflow/meta/online_rus_boost.py [hunk 6]
```diff
@@ -229,20 +232,24 @@ def partial_fit(self, X, y, classes=None, sample_weight=None):
...
                     if y[j] == 1:
                         lam_rus = ((lam * self.n_pos) / (self.n_pos + self.n_neg)) / \
                                   (self.lam_pos[i] / (self.lam_pos[i] + self.lam_neg[i]))
                     else:
-                        lam_rus = ((lam * self.sampling_rate * self.n_pos) / (self.n_pos + self.n_neg)) / \
-                                  (self.lam_neg[i] / (self.lam_pos[i] + self.lam_neg[i]))
+                        lam_rus = ((lam * self.sampling_rate * self.n_pos) / (self.n_pos +
+                                                                              self.n_neg)) / (
+                                              self.lam_neg[i] / (
+                                                  self.lam_pos[i] + self.lam_neg[i]))
```
Comment (smastelini) [inline src/skmultiflow/meta/online_rus_boost.py:252]: The indentation here looks odd to me

File: tox.ini [hunk 7]
```diff
@@ -1,2 +1,3 @@
 [flake8]
-max-line-length=99
\ No newline at end of file
+max-line-length=99
+extend-ignore=E741
```
Comment (smastelini) [inline tox.ini:3]: Is flake8 flagging out violations of this rule?

File: src/skmultiflow/lazy/sam_knn.py [hunk 8]
```diff
@@ -205,22 +207,30 @@ def size_check_fade_out(self):
...
+                                                self._STMLabels[:numShifts])
+                    self._LTMSamples, self._LTMLabels = self.cluster_down(
+                        self._LTMSamples, self._LTMLabels)
                     self._STMSamples = np.delete(self._STMSamples, shiftRange, 0)
                     self._STMLabels = np.delete(self._STMLabels, shiftRange, 0)
-                    self.STMDistances[:len(self._STMLabels),:len(self._STMLabels)] = self.STMDistances[numShifts:len(self._STMLabels)+numShifts, numShifts:len(self._STMLabels)+numShifts]
+                    self.STMDistances[:len(self._STMLabels),
+                    :len(self._STMLabels)] = self.STMDistances[  # noqa: E128
+                                             numShifts:len(self._STMLabels) + numShifts,
+                                             numShifts:len(self._STMLabels) + numShifts]
```
Comment (smastelini) [inline src/skmultiflow/lazy/sam_knn.py:233]: Tough case here. Would using the following strategy exceed the length limit?

`self.STMDistances[:len(self._STMLabels), :len(self._STMLabels)] = \`

File: src/skmultiflow/lazy/sam_knn.py [hunk 9]
```diff
@@ -266,23 +280,31 @@ def _partial_fit(self, x, y):
...
                 oldSTMLabels = self._STMLabels[delrange]
                 self._STMSamples = np.delete(self._STMSamples, delrange, 0)
                 self._STMLabels = np.delete(self._STMLabels, delrange, 0)
-                self.STMDistances[:len(self._STMLabels),:len(self._STMLabels)] = self.STMDistances[(oldWindowSize-newWindowSize):(oldWindowSize-newWindowSize)+len(self._STMLabels),(oldWindowSize-newWindowSize):(oldWindowSize-newWindowSize)+len(self._STMLabels)]
+                diff_window_size = oldWindowSize - newWindowSize
+                self.STMDistances[:len(self._STMLabels),
+                :len(self._STMLabels)] = self.STMDistances[  # noqa: E128
+                                         diff_window_size:diff_window_size + len(self._STMLabels),
+                                         diff_window_size:diff_window_size + len(
+                                             self._STMLabels)]  # noqa: E128
```
Comment (smastelini) [inline src/skmultiflow/lazy/sam_knn.py:307]: Another hard choice :/

File: tox.ini (hunk 7 above)
Comment (garawalid) [reply tox.ini:3]: Here is what I get once I enable that rule:

```console
$ ./dev/linter.sh
src/skmultiflow/evaluation/evaluate_prequential_delayed.py:349:17: E741 ambiguous variable name 'l'
src/skmultiflow/meta/oza_bagging.py:258:25: E741 ambiguous variable name 'l'
src/skmultiflow/meta/oza_bagging.py:270:13: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_smote_bagging.py:346:29: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_smote_bagging.py:362:13: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_boosting.py:339:29: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_boosting.py:356:13: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_adac2.py:363:29: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_adac2.py:379:13: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_under_over_bagging.py:320:29: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_under_over_bagging.py:336:13: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_csb2.py:356:29: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_csb2.py:373:13: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_rus_boost.py:385:29: E741 ambiguous variable name 'l'
src/skmultiflow/meta/online_rus_boost.py:402:13: E741 ambiguous variable name 'l'

```

File: src/skmultiflow/meta/online_boosting.py (hunk 5 above)
Comment (garawalid) [reply src/skmultiflow/meta/online_boosting.py:205]: Nice catch!

File: src/skmultiflow/meta/online_rus_boost.py (hunk 6 above)
Comment (garawalid) [reply src/skmultiflow/meta/online_rus_boost.py:252]: Let me fix it!

File: src/skmultiflow/lazy/sam_knn.py (hunk 8 above)
Comment (garawalid) [reply src/skmultiflow/lazy/sam_knn.py:233]: This way is better: 
```python
                    self.STMDistances[:len(self._STMLabels), :len(self._STMLabels)] = \
                        self.STMDistances[numShifts:len(self._STMLabels) + numShifts,
                                          numShifts:len(self._STMLabels) + numShifts]
```

File: src/skmultiflow/lazy/sam_knn.py (hunk 9 above)
Comment (garawalid) [reply src/skmultiflow/lazy/sam_knn.py:307]: Fixed!

File: src/skmultiflow/meta/classifier_chains.py (hunk 4 above)
Comment (garawalid) [reply src/skmultiflow/meta/classifier_chains.py]: Nice catch! I fixed it!

File: src/skmultiflow/evaluation/base_evaluator.py (hunk 2 above)
Comment (garawalid) [reply src/skmultiflow/evaluation/base_evaluator.py]: I agree! Let me correct it!

**Comment by garawalid** [comment]
@smastelini sorry for the late response! I addressed all your comments.

**Comment by garawalid** [comment]
Thanks for your time @smastelini, I fixed `E741`.

**Comment by smastelini** [comment]
Thank you very much @garawalid for the changes! I'll merge them into `master`.

**Pull request merged**